src/cryptoadvance/specter/notifications/notification_manager.py

```diff
@@ -0,0 +1,164 @@
+import logging, datetime
+
+logger = logging.getLogger(__name__)
+
+from .notifications import Notification
+
+
+class NotificationManager:
+    "Stores and distributes notifications to 1 or more ui_notifications"
+
+    def __init__(self, ui_notifications):
+        """
+        Arguments:
+            - ui_notifications: A list of ui_notifications with the "default" one in position 0 and if (not available), the next ui_notification
+        """
+        self.ui_notifications = ui_notifications
+        self.notifications = []
+
+    def deactivate_target_ui(self, target_ui):
+        for ui_notification in self.ui_notifications:
+            if ui_notification.name == target_ui:
+                ui_notification.is_available = False
+                logger.debug(f"deactivating {ui_notification.name }")
+
+    def show(self, notification):
+        """
+        forwards the notification to ui_notifications, that are in notification['target_uis']
+
+        If a target_ui of notification['target_uis'] is not is_available, then try with the next target_ui
+        """
+        if notification["target_uis"] == ["internal_notification"]:
+            return
+
+        logger.debug(f"show {notification}")
+
+        for ui_notification in self.ui_notifications:
+            if ui_notification.name in notification["target_uis"]:
+
+                notification_shown = ui_notification.show(notification)
```

Comment:
`show` also [checks](https://github.com/relativisticelectron/specter-desktop/blob/f48642ac768603f32f11f6157e87262367149686/src/cryptoadvance/specter/notifications/ui_notifications.py#L34)  if `self.is_available`.  And if this is available, then it is assumed the message was shown.

The flow of removing the permission of the Notifications_API (or any UINotifications) is the following:

1. The `shown` can be either guaranteed (like a print or logging) and immediately given back.  If the notification is somehow async, then `WebAPINotifications` deals with it in the following way:
2. `WebAPINotifications` sends it to javascript and returns notification_shown = `True`
3. If it is not shown  (because [permission](https://github.com/relativisticelectron/specter-desktop/blob/f48642ac768603f32f11f6157e87262367149686/src/cryptoadvance/specter/templates/includes/notifications.js#L66) not `granted`) javascript will retry, and if permission is `denied`, will send [notification_webapi_notification_unavailable](https://github.com/relativisticelectron/specter-desktop/blob/f48642ac768603f32f11f6157e87262367149686/src/cryptoadvance/specter/templates/includes/notifications.js#L66) to python. This then will set `is_available = False` and rebroadcast.

Reply:
I renamed notification_shown to notification_broadcasted. It was very confusing also for me.

Reply:
I now made it constent across al UIs and I save the last_shown_date for each target_ui:  https://github.com/cryptoadvance/specter-desktop/pull/1766/commits/1bf9cbea1d3d476ce68d13689d03d859df31e6b9